### PR TITLE
[IMP] business_requirement: Improve BR module based on Task 13885 [OC…

### PR DESCRIPTION
**Pull request by martintorresvm**
…A/project] [ADD] gap_analysis (#127), add module "script_business_requirement_category" which contains erppeek script to migrate category information

cc @duanyp1991 @elicoidal @xie8899
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63651795%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63652356%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63655336%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63814204%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63815147%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63821145%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63826349%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63990755%22%2C%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63990764%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%203e5753b4e210a0883d2073c985f19a2f2980e718%20business_requirement/models/business.py%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63990764%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22s/business_requirement_categ_id/business_requirement_categ_ids%22%2C%20%22created_at%22%3A%20%222016-05-20T04%3A46%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement/models/business.py%3AL63-77%22%7D%2C%20%22Pull%20285256d3035c1ee5d625d54237aa7320afd75aa2%20script_business_requirement_category/README.rst%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63651795%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20very%20specific%20to%20the%20current%20situation%3A%20I%20think%20this%20part%20is%20not%20necessary%20in%20this%20repo.%22%2C%20%22created_at%22%3A%20%222016-05-18T06%3A32%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40elicoidal%20Do%20you%20mean%20for%20this%20occasion%20is%20not%20necessary%20a%20README.rst%20file%3F%20Should%20I%20remove%20it%3F%22%2C%20%22created_at%22%3A%20%222016-05-18T06%3A38%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16461806%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/victormartinelicocorp%22%7D%7D%2C%20%7B%22body%22%3A%20%22No%20README%20is%20ok%3A%20it%20is%20about%20the%20script%20itself%20which%20is%20not%20part%20of%20the%20module%20%28this%20is%20a%20one-time%20script%20for%20a%20module%20in%20WIP%20so%20it%20is%20not%20like%20it%20was%20in%20production%20in%20many%20companies%29%22%2C%20%22created_at%22%3A%20%222016-05-18T07%3A11%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40elicoidal%20which%20repository%20should%20I%20push%20the%20folder%20with%20the%20script%3F%22%2C%20%22created_at%22%3A%20%222016-05-19T02%3A20%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16461806%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/victormartinelicocorp%22%7D%7D%2C%20%7B%22body%22%3A%20%22elico_odoo%22%2C%20%22created_at%22%3A%20%222016-05-19T02%3A34%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40elicoidal%20done%2C%20I%20removed%20the%20folder%20which%20contains%20the%20script%22%2C%20%22created_at%22%3A%20%222016-05-19T04%3A19%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16461806%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/victormartinelicocorp%22%7D%7D%2C%20%7B%22body%22%3A%20%22great%22%2C%20%22created_at%22%3A%20%222016-05-19T06%3A04%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20script_business_requirement_category/README.rst%3AL1-49%22%7D%2C%20%22Pull%203e5753b4e210a0883d2073c985f19a2f2980e718%20business_requirement/models/business.py%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/50%23discussion_r63990755%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22remove%20since%20this%20is%20not%20a%20%2A%2Afunction%2A%2A%20field%22%2C%20%22created_at%22%3A%20%222016-05-20T04%3A46%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement/models/business.py%3AL63-77%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 285256d3035c1ee5d625d54237aa7320afd75aa2 script_business_requirement_category/README.rst 1'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/50#discussion_r63651795'>File: script_business_requirement_category/README.rst:L1-49</a></b>
- <a href='https://github.com/elicoidal'></a> This is very specific to the current situation: I think this part is not necessary in this repo.
- <a href='https://github.com/victormartinelicocorp'></a> @elicoidal Do you mean for this occasion is not necessary a README.rst file? Should I remove it?
- <a href='https://github.com/elicoidal'></a> No README is ok: it is about the script itself which is not part of the module (this is a one-time script for a module in WIP so it is not like it was in production in many companies)
- <a href='https://github.com/victormartinelicocorp'></a> @elicoidal which repository should I push the folder with the script?
- <a href='https://github.com/elicoidal'></a> elico_odoo
- <a href='https://github.com/victormartinelicocorp'></a> @elicoidal done, I removed the folder which contains the script
- <a href='https://github.com/elicoidal'></a> great
- [ ] <a href='#crh-comment-Pull 3e5753b4e210a0883d2073c985f19a2f2980e718 business_requirement/models/business.py 9'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/50#discussion_r63990755'>File: business_requirement/models/business.py:L63-77</a></b>
- <a href='https://github.com/duanyp1991'></a> remove since this is not a **function** field
- [ ] <a href='#crh-comment-Pull 3e5753b4e210a0883d2073c985f19a2f2980e718 business_requirement/models/business.py 4'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/50#discussion_r63990764'>File: business_requirement/models/business.py:L63-77</a></b>
- <a href='https://github.com/duanyp1991'></a> s/business_requirement_categ_id/business_requirement_categ_ids

<a href='https://www.codereviewhub.com/xie8899/project/pull/50?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/xie8899/project/pull/50?approve=1'></a>&nbsp;<a href='https://github.com/xie8899/project/pull/50'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: script_business_requirement_category/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline script_business_requirement_category/README.rst]: This is very specific to the current situation: I think this part is not necessary in this repo.

Comment (martintorresvm) [inline script_business_requirement_category/README.rst]: @elicoidal Do you mean for this occasion is not necessary a README.rst file? Should I remove it?

Comment (elicoidal) [inline script_business_requirement_category/README.rst]: No README is ok: it is about the script itself which is not part of the module (this is a one-time script for a module in WIP so it is not like it was in production in many companies)

Comment (martintorresvm) [inline script_business_requirement_category/README.rst]: @elicoidal which repository should I push the folder with the script?

Comment (elicoidal) [inline script_business_requirement_category/README.rst]: elico_odoo

Comment (martintorresvm) [inline script_business_requirement_category/README.rst]: @elicoidal done, I removed the folder which contains the script

Comment (elicoidal) [inline script_business_requirement_category/README.rst]: great

File: business_requirement/models/business.py [hunk 2]
```diff
@@ -63,6 +63,14 @@ def _get_default_company(self):
         store=True,
         states={'draft': [('readonly', False)]}
     )
+    business_requirement_categ_id = fields.Many2many(
+        'business.requirement.category',
+        string='Business Requirement Categ',
+        relation='business_requirement_category_rel',
+        readonly=True,
+        store=True,
```
Comment (duanyp1991) [inline business_requirement/models/business.py:71]: remove since this is not a **function** field

File: business_requirement/models/business.py [hunk 3]
```diff
@@ -63,6 +63,14 @@ def _get_default_company(self):
         store=True,
         states={'draft': [('readonly', False)]}
     )
+    business_requirement_categ_id = fields.Many2many(
```
Comment (duanyp1991) [inline business_requirement/models/business.py:66]: s/business_requirement_categ_id/business_requirement_categ_ids

**Pull request closed without merging**